### PR TITLE
not always istio and not always grpc

### PR DESCRIPTION
**Pull request by jkminneti**

### CODE REVIEW

File: pkg/core/init.go [hunk 1]
```diff
@@ -9,50 +9,55 @@ import (
...
-		return
+	err = createServiceYAML(appName, serviceType)
+	if err != nil {
+		return err
 	}
 
-	if err = createVirtualServiceYAML(appName, routePrefix); err != nil {
-		return
+	if !withIstio {
+		return nil
```
Comment (jkminneti) [inline pkg/core/init.go]: this and

File: pkg/core/init.go [hunk 2]
```diff
@@ -9,50 +9,55 @@ import (
...
 
 	return writeYAML(data.Deployment, templateData)
 }
 
-func createServiceYAML(appName string) (err error) {
+func createServiceYAML(appName string, serviceType string) error {
 	templateData := map[string]string{
-		"appName": appName,
+		"appName":     appName,
+		"serviceType": serviceType,
```
Comment (jkminneti) [inline pkg/core/init.go]: ..this are the only actual changes on this file, otherwise it's just de-naked-returnification

File: cmd/init.go [hunk 3]
```diff
@@ -16,13 +17,40 @@ var initCmd = &cobra.Command{
...
+	"redis":    true,
+	"tcp":      true,
+	"tls":      true,
+	"udp":      true,
+}
+
+func initialize(cmd *cobra.Command, args []string) error {
 	appName := args[0]
 	routePrefix := args[1]
+	serviceType := args[2]
```
Comment (Jefferson-Faseler) [inline cmd/init.go]: Are these args or descriptions out of date?

Comment (Jefferson-Faseler) [reply cmd/init.go]:
```
go run main.go init pine apple --istio=grpc
panic: runtime error: index out of range [2] with length 2

goroutine 1 [running]:
tuber/cmd.initialize(0x1dba5a0, 0xc0002ce450, 0x2, 0x3, 0x0, 0x0)
	/Users/jeffersonfaseler/freshly/tuber/cmd/init.go:39 +0x1e2
github.com/spf13/cobra.(*Command).execute(0x1dba5a0, 0xc0002ce390, 0x3, 0x3, 0x1dba5a0, 0xc0002ce390)
	/Users/jeffersonfaseler/.go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0x1dba820, 0x0, 0x0, 0x0)
	/Users/jeffersonfaseler/.go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/jeffersonfaseler/.go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
tuber/cmd.Execute(0x9a400, 0xc0000a8058)
	/Users/jeffersonfaseler/freshly/tuber/cmd/root.go:54 +0x42
main.main()
	/Users/jeffersonfaseler/freshly/tuber/main.go:8 +0x22
exit status 2
```

File: data/tuberapps/service.go [hunk 4]
```diff
@@ -4,5 +4,5 @@ package data
 // Service is generated. Returns the default service for a new tuber app
 var Service = TuberYaml{
 	Filename: "service.yaml",
-	Contents: []byte{ 97,112,105,86,101,114,115,105,111,110,58,32,118,49,10,107,105,110,100,58,32,83,101,114,118,105,99,101,10,109,101,116,97,100,97,116,97,58,10,32,32,110,97,109,101,58,32,123,123,46,97,112,112,78,97,109,101,125,125,10,115,112,101,99,58,10,32,32,112,111,114,116,115,58,10,32,32,45,32,112,111,114,116,58,32,51,48,48,48,10,32,32,32,32,110,97,109,101,58,32,103,114,112,99,10,32,32,115,101,108,101,99,116,111,114,58,10,32,32,32,32,97,112,112,58,32,123,123,46,97,112,112,78,97,109,101,125,125,10, },
+	Contents: []byte{ 97,112,105,86,101,114,115,105,111,110,58,32,118,49,10,107,105,110,100,58,32,83,101,114,118,105,99,101,10,109,101,116,97,100,97,116,97,58,10,32,32,110,97,109,101,58,32,123,123,46,97,112,112,78,97,109,101,125,125,10,115,112,101,99,58,10,32,32,112,111,114,116,115,58,10,32,32,45,32,112,111,114,116,58,32,51,48,48,48,10,32,32,32,32,110,97,109,101,58,32,123,123,46,115,101,114,118,105,99,101,84,121,112,101,125,125,10,32,32,115,101,108,101,99,116,111,114,58,10,32,32,32,32,97,112,112,58,32,123,123,46,97,112,112,78,97,109,101,125,125,10, },
```
Comment (Jefferson-Faseler) [inline data/tuberapps/service.go]: Assuming this is a json/yaml template?

File: data/tuberapps/yamls/service.yaml [hunk 5]
```diff
@@ -5,6 +5,6 @@ metadata:
 spec:
   ports:
   - port: 3000
-    name: grpc
+    name: {{.serviceType}}
```
Comment (Jefferson-Faseler) [inline data/tuberapps/yamls/service.yaml]: 🙇

**Pull request merged**